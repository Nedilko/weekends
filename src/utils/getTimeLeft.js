import dayjs from 'dayjs'
import objectSupport from 'dayjs/plugin/objectSupport'
dayjs.extend(objectSupport)

const FRIDAY = {
  day: 4,
  hour: 17,
  minute: 59,
  second: 59,
}

const SECONDS_IN = {
  day: 86400,
  hour: 3600,
  minute: 60,
}

const getTimeLeft = (friday = FRIDAY) => {
  const secondsLeft = dayjs().set(friday).diff(dayjs(), 'second')
  if (secondsLeft <= 0) {
    return {
      days: 0,
      hours: 0,
      minutes: 0,
      seconds: 0,
    }
  }
  const days = Math.trunc(secondsLeft / SECONDS_IN.day)
  const hours = Math.trunc(secondsLeft / SECONDS_IN.hour)
  const minutes = Math.trunc(secondsLeft / SECONDS_IN.minute)
  return {
    days,
    hours: hours - days * 24,
    minutes: minutes - hours * 60,
    seconds: secondsLeft - minutes * 60,
  }
}

const isFriday = (friday) => Object.values(friday).every((value) => value === 0)

export { getTimeLeft, isFriday }
