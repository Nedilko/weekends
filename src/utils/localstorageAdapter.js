const writeToLocalstorage = (key, value) => {
  try {
    localStorage.setItem(key, value)
  } catch (e) {
    console.error(e)
  }
}

const readFromLocalstorage = (key) => {
  try {
    return localStorage.getItem(key)
  } catch (e) {
    console.error(e)
  }
}

export { readFromLocalstorage, writeToLocalstorage }
