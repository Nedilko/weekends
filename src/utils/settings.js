const DEAFULT_FRIDAY = {
  day: 5,
  hour: 17,
  minute: 59,
  second: 59,
}

const DEFAULT_THEME = 'light'

const getDefaultSettings = () => ({
  greetingsText: 'Have a beer!',
  day: DEAFULT_FRIDAY.day,
  hour: DEAFULT_FRIDAY.hour,
  theme: DEFAULT_THEME,
  useSystemTheme: false,
})

const getTimerData = (friday) => ({
  days: friday.day,
  hours: friday.hour,
  minutes: DEAFULT_FRIDAY.minute,
  seconds: DEAFULT_FRIDAY.second,
})

export { DEAFULT_FRIDAY, getTimerData, getDefaultSettings }
