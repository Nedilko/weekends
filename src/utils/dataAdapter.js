import { getDefaultSettings } from '@utils/settings'
import {
  writeToLocalstorage,
  readFromLocalstorage,
} from '@utils/localstorageAdapter'

const loadSettings = () => {
  const localStorData = readFromLocalstorage('settings')
  if (!localStorData) {
    writeSettings(getDefaultSettings())
    return getDefaultSettings()
  }
  return JSON.parse(atob(localStorData))
}

const writeSettings = (settings) => {
  writeToLocalstorage('settings', btoa(JSON.stringify(settings)))
}

export { loadSettings, writeSettings }
