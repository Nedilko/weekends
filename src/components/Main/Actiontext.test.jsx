import Actiontext from './ActionText'
import { render, screen } from '../../utils/test-utils'

describe('text in the document', () => {
  it('action text is visible', () => {
    render(<Actiontext />)
    const title = screen.getByRole('heading')
    expect(title).toBeInTheDocument()
    expect(title).toMatchSnapshot()
  })
})
