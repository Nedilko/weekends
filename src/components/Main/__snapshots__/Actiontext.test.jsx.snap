// Vitest Snapshot v1

exports[`text in the document > action text is visible 1`] = `
<h2
  class="text-center text-3xl font-thin uppercase"
>
  Time left to weekends
</h2>
`;

exports[`text in the document > snapshot 1`] = `
<h2
  class="text-center text-3xl font-thin uppercase"
>
  Time left to weekends
</h2>
`;
